### DISCUSSION TITLE
Y2 or M limit stop to work after upgraded.

### DISCUSSION THREAD

**Discussion by rmaltais** [grblHAL/core#518] [Q&A] [answered]
Hi

Since last version, i lost my signal on my Y2 limit switch. Motor work but i can't square my Y axe because only one limit work.

I reinstall old version and it's work fine. It's not a hardware problem because if i inverse my sensor, it's work fine on only INPUT3 on my card.

This is my setting and version now it is work properly:

`[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:STM32F103RC]
[DRIVER VERSION:230526]
[BOARD:MACH3 USB BOB (BSMCE04U)]`

thank

**Comment by terjeio** [comment]
What is the `$LIM` output with Y2 triggered and not triggered?
And is the Y2 input listed in the `$PINS` output?

**Reply by rmaltais** [reply]
Here when triggered or not

$LIM
[LIMITS:,,,]
$LIM
[LIMITS:,,,]

Here OUTPUT $PINS

$PINS
[PIN:PB8,Feed hold]
[PIN:PB7,Cycle start]
[PIN:PB9,Emergency stop]
[PIN:PB4,Probe]
[PIN:PB12,X limit min]
[PIN:PB13,Y limit min]
[PIN:PB14,Z limit min]
[PIN:PB6,Aux in 0,P0]
[PIN:PB3,Aux in 1,P1]
[PIN:PB1,X step]
[PIN:PC5,Y step]
[PIN:PA5,Y2 step]
[PIN:PA7,Z step]
[PIN:PB0,X dir]
[PIN:PC4,Y dir]
[PIN:PA3,Y2 dir]
[PIN:PA6,Z dir]
[PIN:PB10,Steppers enable]
[PIN:PC6,Spindle on]
[PIN:PC7,Spindle direction]
[PIN:PA8,Spindle PWM]
[PIN:PC9,Mist]
[PIN:PC8,Flood]
[PIN:PD2,RX,UART1]
[PIN:PC12,TX,UART1]

**Reply by terjeio** [reply]
Odd, there is no Y2 limit pin in the pins list. Are you compiling locally or via the Web Builder?

**Reply by rmaltais** [reply]
Web Builder

**Reply by terjeio** [reply]
Ok, everything seems ok when I build locally. I cannot verify if the binary is working (I am away) so I leave it to you to try it:

[Firmware.zip](https://github.com/grblHAL/core/files/15390946/Firmware.zip)

If this is working then there is something amiss with the Web Builder.

**Reply by rmaltais** [reply]
> Ok, everything seems ok when I build locally. I cannot verify if the binary is working (I am away) so I leave it to you to try it:
> 
> 
> 
> [Firmware.zip](https://github.com/grblHAL/core/files/15390946/Firmware.zip)
> 
> 
> 
> If this is working then there is something amiss with the Web Builder.

I tried this firmware today. I have a error message about a line of G-Code et iosender close immediately.

![image](https://github.com/grblHAL/core/assets/41405404/87942cc3-0a13-4aa9-8ad8-991f1bdcfb7a)

**Reply by terjeio** [reply]
Can you try connecting with a terminal program such as puTTY?
If that works I want to know if the Y2 limit pin shows up in the $pins output.

**Reply by rmaltais** [reply]
Yes, i can connect via Putty. I have no result on output port. I can see my sensor led up but no result on terminal.

ok
[LIMITS:,,,]
ok
[LIMITS:,,,]
ok

**Reply by terjeio** [reply]
And the $pins output is?

**Reply by rmaltais** [reply]
[PIN:PB8,Feed hold]
[PIN:PB7,Cycle start]
[PIN:PB9,Emergency stop]
[PIN:PB4,Probe]
[PIN:PB12,X limit min]
[PIN:PB13,Y limit min]
[PIN:PB14,Z limit min]
[PIN:PB6,Aux in 0,P0]
[PIN:PB3,Aux in 1,P1]
[PIN:PB1,X step]
[PIN:PC5,Y step]
[PIN:PA5,Y2 step]
[PIN:PA7,Z step]
[PIN:PB0,X dir]
[PIN:PC4,Y dir]
[PIN:PA3,Y2 dir]
[PIN:PA6,Z dir]
[PIN:PB10,Steppers enable]
[PIN:PC6,Spindle on]
[PIN:PC7,Spindle direction]
[PIN:PA8,Spindle PWM]
[PIN:PC9,Mist]
[PIN:PC8,Flood]
[PIN:PD2,RX,UART1]
[PIN:PC12,TX,UART1]

**Comment by terjeio** [comment, marked as answer]
There was something amiss in the Web Builder, it was not possible to build with an auto squared axis. Fixed now so hopefully it works.
If it does not this issue has to wait until I am back home since I do not have access to a board for testing.

**Reply by rmaltais** [reply]
yes, I also found it strange that the button became grayed out when “Auto-Square” was selected. This is difficult to understand and I would like to soon be able to download the generated code to see possible errors. I'll try it this weekend if it works.

**Reply by rmaltais** [reply]
I tried with a new  Web Builder and it's ok. All limit Y and Y2 work perfectly now and auto-squaring too.

Thank for all
